src/Albany_Application.cpp

```diff
@@ -2922,6 +2996,7 @@ Application::evaluateStateFieldManager(
         sfm[ps]
             ->setKokkosExtendedDataTypeDimensions<PHAL::AlbanyTraits::Jacobian>(
                 derivative_dimensions);
+
```

Comment:
I think you need `setSideSetExtents<EvalT>(sfm[ps]);` - i.e. let's go ahead and call this before all fm postReg, even if sides are not currently used. It should skip everything except the final check to make sure all dynamic layouts are set.